### DISCUSSION TITLE
Displaced qp coordinates available in a material? (specific to AbaqusUMATStress)

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#22835] [Q&A Modules: Solid mechanics] [answered]
In `AbaqusUMATStress`, the coordinates of the quadrature points on the _undisplaced_ mesh are passed in [here](https://github.com/idaholab/moose/blob/1e9400c0f226ae3f9118aa43dd4c4217df242a90/modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C#L217-L219). To match Abaqus, these would need to be the coordinates on the _displaced_ mesh. Is this information available in this particular Material? Maybe the displacements (interpolated to the qp) can be determined based on other available quantities?

@dschwen @recuero @GiudGiud

**Comment by hugary1995** [comment, marked as answer]
Dumb question, use_displaced_mesh = true doesn't work here, does it?

**Reply by GiudGiud** [reply]
if it doesnt, displacements are easy to get from
```
params.addRequiredCoupledVar("displacements", "disp variables");

_disp(coupledValues("displacements")),
```
then you can displace or un-displace at will

But it should work (and if not, I m curious where I am wrong below). 
The way it happens is: use_displaced_mesh is set to true in the material params
the FEProblem sees this and sets the material's problem as the displacedProblem
```
  if (_displaced_problem && parameters.get<bool>("use_displaced_mesh"))
  {
    parameters.set<SubProblem *>("_subproblem") = _displaced_problem.get();
    _reinit_displaced_elem = true;
  }
```
the displacedProblem holds a displaced system. The displaced system holds the assembly.
The q_points are retrieved from the assembly through a FEBase then a FEMap which ultimately gets it from the mesh. 
The mesh is the displaced mesh as the displaced system gets the mesh from the displaced problem.
/* end deep dive, I was curious */

**Reply by jessecarterMOOSE** [reply]
@hugary1995 Wish it were that easy, but no dice. 

```
*** ERROR ***
The parameter 'Materials/umat/use_displaced_mesh' is a private parameter and should not be used in an input file.
```

Let me try what @GiudGiud suggested.

**Reply by hugary1995** [reply]
We probably should unsuppress the use_displaced_mesh parameter in AbaqusUMATStress. @dschwen @recuero what do you think?

**Reply by jessecarterMOOSE** [reply]
Looks like it's suppressed in `ComputeStressBase`, which `AbaqusUMATStress` inherits from, which throws an error if set.

```
  if (getParam<bool>("use_displaced_mesh"))                                                                                                                                                    
    mooseError("The stress calculator needs to run on the undisplaced mesh.");
```

Not sure if that's supposed to be carried to derived classes.

**Reply by jessecarterMOOSE** [reply]
Going with @GiudGiud 's first suggestion. Will submit a PR...

**Reply by recuero** [reply]
I _believe_ that should work. May need to regold some tests.
